src/Sulu/Bundle/ContentBundle/Behat/ContentContext.php

```diff
@@ -0,0 +1,159 @@
+        <title lang="en">%s</title>
+    </meta>
+
+    <properties>
+        <property name="title" type="text_line" mandatory="true">
+            <meta>
+                <title lang="de">Tier</title>
+                <title lang="en">Animals</title>
+            </meta>
+
+            <tag name="sulu.rlp.part"/>
+        </property>
+
+        <property name="url" type="resource_locator" mandatory="true">
+            <meta>
+                <title lang="de">Adresse</title>
+                <title lang="en">Resourcelocator</title>
+            </meta>
+
+            <tag name="sulu.rlp"/>
+        </property>
+
+%s
+    </properties>
+</template>
+EOT
+        ;
+        $template = sprintf($template,
+            $arg1, $arg1, $arg1, $arg1, $string->getRaw()
+        );
+
+        $this->createStructureTemplate('page', $arg1, $template);
+    }
+
+    /**
+     * @Given the following pages exist:
+     */
+    public function theFollowingPagesExist(TableNode $table)
+    {
+        $data = $table->getColumnsHash();
+
+        $this->createStructures('page', $data);
+    }
+
+    /**
+     * @Given I am editing a page of type :arg1
+     */
+    public function iAmEditingAPageOfType($arg1)
+    {
+        $request = ContentMapperRequest::create()
+            ->setTemplateKey($arg1)
+            ->setType('page')
+            ->setWebspaceKey('sulu_io')
+            ->setUserId($this->getUserId())
+            ->setState(StructureInterface::STATE_PUBLISHED)
+            ->setLocale('de')
+            ->setData(array(
+                'title' => 'Behat Test Content',
+                'url' => '/behat-test-content',
+            ));
+
+        $page = $this->getContentMapper()->saveRequest($request);
+        $this->visitPath('/admin/#content/contents/sulu_io/de/edit:' . $page->getUuid() . '/content');
+        $this->getSession()->wait(5000, '$("#content-form").length');
+        sleep(1); // wait one more second to avoid flaky tests
+    }
+
+    /**
+     * @Then I should see the date picker
+     */
+    public function iShouldSeeTheDatePicker()
+    {
+        return $this->getSession()->evaluateScript('$(".datepicker").length');
+    }
+
+    /**
+     * @Then I should see the color picker
+     */
+    public function iShouldSeeTheColorPicker()
+    {
+        return $this->getSession()->evaluateScript('$(".minicolors").length');
+    }
+
+    /**
+     * @Then I click the first smart content filter icon
+     */
+    public function iClickTheFirstSmartContentFilterIcon()
+    {
+        $this->clickSelector('.smart-content-container .fa-filter');
+    }
+
+    /**
+     * @Then I expect the smart content overlay to slide right
+     */
+    public function iWaitForTheSmartContentOverlaytoSlideRight()
+    {
+        $this->getSession()->executeScript(<<<EOT
+document.smartContentEvent = false;
+app.sandbox.on('husky.overlay.smart-content.smart_content.slide-right', function () { document.smartContentEvent = true; });
```

Comment:
I thought I have already seen a context containing a function which waits for an event. You could just wait for the `'husky.overlay.smart-content.smart_content.slide-right'`-event, can't you?